plugins/discovery-ec2/src/test/java/org/opensearch/discovery/ec2/ConfigPathSupport.java

```diff
@@ -0,0 +1,32 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.discovery.ec2;
+
+import org.opensearch.common.SuppressForbidden;
+import org.opensearch.common.io.PathUtils;
+
+import java.nio.file.Path;
+
+/**
+ * The trait that adds the config path and AWS profile setup to the test cases.
+ */
+interface ConfigPathSupport {
+    default Path configPath() {
+        return PathUtils.get("config");
+    }
+
+    @SuppressForbidden(reason = "set predictable aws defaults")
+    default public void setUpAwsProfile() throws Exception {
+        SocketAccess.doPrivileged(() -> System.setProperty("opensearch.path.conf", configPath().toString()));
```

Comment:
This `opensearch.path.conf` manipulation could bite us -  we run tests in parallel (multiple forks) and the property change affect the whole forked JVM, the tests run after this one will see a new `opensearch.path.conf` setting, we at least should reset it back after each test case.

Reply:
Done.